src/main/java/frc/robot/subsystems/intake/IntakeIOTalonFX.java

```diff
@@ -0,0 +1,200 @@
+    drumIRSensor = new DigitalInput(IntakeConstants.INTAKE_DRUM_IR_SENSOR_ID);
+
+    rightRollerVelocityRequest = new VelocityTorqueCurrentFOC(0);
+    leftRollerVelocityRequest = new VelocityTorqueCurrentFOC(0);
+    drumVelocityRequest = new VelocityTorqueCurrentFOC(0);
+
+    rightRollerVelocityStatusSignal = rightRollerMotor.getRotorVelocity();
+    leftRollerVelocityStatusSignal = leftRollerMotor.getRotorVelocity();
+    drumVelocityStatusSignal = drumMotor.getRotorVelocity();
+
+    rightRollerStatorCurrentStatusSignal = rightRollerMotor.getStatorCurrent();
+    leftRollerStatorCurrentStatusSignal = leftRollerMotor.getStatorCurrent();
+    drumStatorCurrentStatusSignal = drumMotor.getStatorCurrent();
+
+    rightRollerSupplyCurrentStatusSignal = rightRollerMotor.getSupplyCurrent();
+    leftRollerSupplyCurrentStatusSignal = leftRollerMotor.getSupplyCurrent();
+    drumSupplyCurrentStatusSignal = drumMotor.getSupplyCurrent();
+
+    configureIntakeRollerMotors(
+        IntakeConstants.INTAKE_RIGHT_ROLLER_MOTOR_ID, IntakeConstants.INTAKE_LEFT_ROLLER_MOTOR_ID);
+    configureIntakeDrumMotor(IntakeConstants.INTAKE_DRUM_MOTOR_ID);
+  }
+
+  @Override
+  public void updateInputs(IntakeIOInputs inputs) {
+    BaseStatusSignal.refreshAll(
+        rightRollerVelocityStatusSignal,
+        leftRollerVelocityStatusSignal,
+        drumVelocityStatusSignal,
+        rightRollerStatorCurrentStatusSignal,
+        leftRollerStatorCurrentStatusSignal,
+        drumStatorCurrentStatusSignal,
+        rightRollerSupplyCurrentStatusSignal,
+        leftRollerSupplyCurrentStatusSignal,
+        drumSupplyCurrentStatusSignal);
+
+    inputs.isRightRollerIRBlocked = rightRollerIRSensor.get();
+    inputs.isLeftRollerIRBlocked = leftRollerIRSensor.get();
+    inputs.isDrumIRBlocked = drumIRSensor.get();
+
+    inputs.rightRollerStatorCurrentAmps = rightRollerStatorCurrentStatusSignal.getValueAsDouble();
+    inputs.leftRollerStatorCurrentAmps = leftRollerStatorCurrentStatusSignal.getValueAsDouble();
+    inputs.drumStatorCurrentAmps = drumStatorCurrentStatusSignal.getValueAsDouble();
+
+    inputs.rightRollerSupplyCurrentAmps = rightRollerSupplyCurrentStatusSignal.getValueAsDouble();
+    inputs.leftRollerSupplyCurrentAmps = leftRollerSupplyCurrentStatusSignal.getValueAsDouble();
+    inputs.drumSupplyCurrentAmps = drumSupplyCurrentStatusSignal.getValueAsDouble();
+
+    inputs.rightRollerVelocityRotationsPerSecond =
+        rightRollerVelocityStatusSignal.getValueAsDouble();
+    inputs.leftRollerVelocityRotationsPerSecond = leftRollerVelocityStatusSignal.getValueAsDouble();
+    inputs.drumVelocityRotationsPerSecond = drumVelocityStatusSignal.getValueAsDouble();
+
+    inputs.rightRollerReferenceVelocityRPS = rightRollerRequestedVelocity;
+    inputs.leftRollerReferenceVelocityRPS = leftRollerRequestedVelocity;
+    inputs.drumReferenceVelocityRPS = drumRequestedVelocity;
+  }
```

Comment:
You also need to check all your tunables in this method and update the configuration if any values change. Look at DrivetrainIOCTRE's updateInputs method for an example.